.github/workflows/build_wheels.yml

```diff
@@ -98,92 +94,158 @@ jobs:
-            pip install --pre -i https://pypi.anaconda.org/scipy-wheels-nightly/simple scipy
-          CIBW_ENVIRONMENT: CXX=c++ NETWORKIT_OSX_CROSSBUILD=ON
-          CIBW_ARCHS_MACOS: ${{ matrix.cibw_archs }}
-          CIBW_BUILD: "cp39-*"
+          CIBW_BEFORE_BUILD: pip install cython ipython
+          CIBW_ARCHS: "AMD64"
+          CIBW_BUILD: "cp38-* cp39-* cp310-*"
           CIBW_SKIP: pp*
       - uses: actions/upload-artifact@v2
         with:
-          name: networkit-cp39-macos-${{ matrix.cibw_archs }}.whl
+          path: ./wheelhouse/*cp38*.whl
+          retention-days: ${{ env.ARTIFACT_RETENTION }}
+      - uses: actions/upload-artifact@v2
+        with:
           path: ./wheelhouse/*cp39*.whl
           retention-days: ${{ env.ARTIFACT_RETENTION }}
+      - uses: actions/upload-artifact@v2
+        with:
+          path: ./wheelhouse/*cp310*.whl
+          retention-days: ${{ env.ARTIFACT_RETENTION }}
 
-  cpython-windows:
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [windows-latest]
+  source-distribution:
+    name: 'Source distribution'
+    runs-on: ubuntu-20.04
     steps:
       - uses: actions/checkout@v2
         with:
           submodules: true
-      - name: Setup devCmd (vstools)
-        uses: ilammy/msvc-dev-cmd@v1
       - uses: actions/setup-python@v2
         name: Install Python
         with:
           python-version: '3.9'
-      - name: Build wheels
-        uses: pypa/cibuildwheel@v2.1.2
-        env:
-          CIBW_BEFORE_BUILD: pip install cython ipython
-          CIBW_ARCHS: "AMD64"
-          CIBW_BUILD: "cp39-*"
-          CIBW_SKIP: pp*
+      - name: Create sdist source
+        run: |
+          pip install cython
+          python3 setup.py build_ext
+          python3 setup.py sdist
       - uses: actions/upload-artifact@v2
         with:
-          name: networkit-cp39-win-amd64.whl
-          path: ./wheelhouse/*cp39*.whl
-          retention-days: ${{ env.ARTIFACT_RETENTION }}
\ No newline at end of file
+          path: ./dist/*networkit*.tar.gz
+          retention-days: ${{ env.ARTIFACT_RETENTION }}
+
+  # This job checks whether the current workflow is triggered by a new tag with the following form: INT.INT or INT.INT.INT (for example: 1.0, 1.0.1, ...)
+  # The result is saved in a variable and used as a conditional variable when uploading packages.
+  check_tag:
+    name: "Check for new release"
+    runs-on: ubuntu-latest
+    outputs:
+      run_other_jobs: ${{ steps.check_tag.outputs.run_jobs }}
+    steps:
+      - name: check tag ${{ github.ref }}
+        id: check-tag
+        run: |
+          if [[ ${{ github.ref }} =~ ^refs\/tags\/release\/[0-9]+\.[0-9]+(\.[0-9]+)?$ ]]; then
+            echo "::set-output name=run_jobs::true"
+          else
+            echo "::set-output name=run_jobs::false"
+          fi
+
+  upload-packages:
+    if: needs.check_tag.outputs.run_other_jobs == 'true' && github.repository == 'networkit/networkit'
```

Comment:
`run_other_jobs` does not match the output of `echo` (which is `run_jobs`), so that won't work here. Maybe something like `is_release` is more appropriate anyway.